File: backend/src/main/java/com/rungo/api/global/security/SecurityUser.java
package com.rungo.api.global.security;

import com.rungo.api.domain.users.enumtype.Role;
import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

import java.util.Collection;

@Getter
public class SecurityUser extends User { // 인증된 사용자 정보

    private Long id;
    private String email;
    private Role role;

    public SecurityUser(Long id, String email, Role role, Collection<? extends GrantedAuthority> authorities) {
        super(email, "password", authorities); // 인증된 사용자의 password값은 필요 없으므로, 문자열 고정
        this.id = id;
        this.email = email;
        this.role = role;
    }
}

File: backend/src/main/java/com/rungo/api/global/security/SecurityUser.kt
package com.rungo.api.global.security

import com.rungo.api.domain.users.enumtype.Role
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User

class SecurityUser(// 인증된 사용자 정보
    val id: Long,
    val email: String,
    val role: Role,
    authorities: Collection<GrantedAuthority>
) : User(email, "password", authorities)